From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MidnightTale <midnighttale24@gmail.com>
Date: Sun, 20 Jul 2025 20:35:00 +0700
Subject: [PATCH] Fix Version Unknown


diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 9699d7dcca5cf67f50ad05c0e875de424a4e00c5..58c0f85972ac159a91a2a03c610c56d04db3c0a0 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/me.earthme.luminol/luminol-api/pom.properties"); // Folia //Luminol
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/fun.mntale.atdeprecated/atdeprecated-api/pom.properties"); // Folia //Luminol // atDeprecated
         Properties properties = new Properties();
 
         if (stream != null) {
